<?xml version="1.0" encoding="ISO-8859-15"?>
<launch>
  <arg name="ns" default="scarab40" />
  
	<!-- Spawn the World -->
  <include file="$(find erl_models)/launch/spawn_world.launch">
    <arg name="world" value="fla_warehouse1"/>
    <arg name="scale" value="1"/>
    <arg name="gui" value="false"/>
  </include>

  <!-- Spawn the Scarab -->
  <include file="$(find erl_models)/launch/spawn_robot.launch">
    <arg name="ns" value="$(arg ns)"/>
    <arg name="tf_prefix" value="$(arg ns)" />
    <arg name="model" value="$(find erl_models)/urdf/scarab/scarab.xacro"/>
    <arg name="px" value="0.0" />
    <arg name="py" value="0.0" />
    <arg name="use_groundtruth_odom" value="true" />
  </include>

  <!-- Run Scarab Localization 
  <include file="$(find erl_models)/launch/scarab/scarab_localization.launch">
    <arg name="ns" value="$(arg ns)"/>
    <arg name="tf_prefix" value="$(arg ns)" />
    <arg name="base_frame" value="base_footprint"/>
    <arg name="laser_frame" value="laser"/>
    <arg name="odom_laser_frame" value="odom"/>
    <arg name="scan_topic" value="scan" />
    <arg name="odom_motor_topic" value="odom" />
    <arg name="odom_laser_topic" value="odom_laser" />
  </include>
  -->
  <!-- Run Scarab Mapping  
  <include file="$(find erl_models)/launch/scarab/scarab_mapping.launch">
    <arg name="ns" value="$(arg ns)"/>
    <arg name="tf_prefix" value="$(arg ns)" />
    <arg name="scan_topic" value="scan" />
    <arg name="base_frame" value="base_footprint"/>
    <arg name="odom_frame" value="odom"/>
    <arg name="map_frame"  value="map"/>        
  </include>
 -->
  <!-- Run Scarab Navigation -->
  <include file="$(find erl_models)/launch/scarab/scarab_navigation.launch">
    <arg name="ns" value="$(arg ns)"/>
    <arg name="tf_prefix" value="$(arg ns)" />
    
    <arg name="map_frame"  value="map"/>
    <arg name="odom_frame" value="odom"/>
    <arg name="base_frame" value="base_footprint"/>
    
    <arg name="map_topic"  value="map"/>
    <arg name="odom_topic" value="odom"/>
    <arg name="scan_topic" value="scan" />
    <arg name="pose_topic" value="pose" />
    <arg name="goal_topic" value="move_base_simple/goal"/>
    
    <arg name="use_hfn" value="true"/>   
  </include>  
 <!-- Run controller in C++ -->
  <node name="power_diagram_controller" pkg="car_control" type="car_control" output="screen">
    <param name="/cmd_vel" value="/scarab40/mobile_base/commands/velocity" />
    <param name="/path" value="/path" />
    <param name="/odom" value="/scarab40/ground_truth/odom" />
    <param name="/kl" value="0.5" />
    <param name="/ka" value="0.1" />
    <param name="/goal_threshold" value="0.1" />
    <param name="/T" value="40" />
    <param name="/vx" value="10.0" />
    <param name="/vy" value="0.0" />
    <param name="/ax" value="0.0" />
    <param name="/ay" value="0.0" />
  </node>
  <!-- Run controller in python
  <node pkg="erl_fastron_mapping" name="scarab_control" type="scarab_control.py" output="screen"/>
    -->
  <!-- Launch rviz -->
  <node pkg="rviz" type="rviz" name="rviz"
        args="-d $(find erl_models)/rviz/scarab40.cfg.rviz"/>
  

</launch>

